packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -31,10 +34,18 @@ export default class UIForm extends React.Component {
 		this.onTrigger = this.onTrigger.bind(this);
 		this.onActionClick = this.onActionClick.bind(this);
 		// control the tv4 language here.
+		const language = getLanguage(props.t);
+		if (typeof props.language === 'function') {
+			Object.assign(language, props.language(props.t));
+		}
 		if (!tv4.language('@talend')) {
-			tv4.addLanguage('@talend', props.language);
+			tv4.addLanguage('@talend', language);
 			tv4.language('@talend'); // set it
 		}
+		const allFormats = Object.assign({}, customFormats(props.t), props.customFormats);
```

Comment:
You can drop the empty object. `customFormats(props.t)` always returns a new object where you can assign new properties.